File: ConnectionJ2EE/MainActivity.java
package com.nejirezgui.connectionj2ee;

import android.content.Intent;
import android.os.AsyncTask;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import org.json.JSONException;
import org.json.JSONObject;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.io.OutputStreamWriter;
import java.net.HttpURLConnection;
import java.net.MalformedURLException;
import java.net.URL;

import javax.net.ssl.HttpsURLConnection;

public class MainActivity extends AppCompatActivity {
    String url_serv = "http://10.0.2.2:8080/ConnectionAndroid/servlet";

    TextView txt_somme;
    EditText ed_nbre1, ed_nbre2;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);


        setContentView(R.layout.activity_main);

        ed_nbre1 = (EditText) findViewById(R.id.ed_nbre1);
        ed_nbre2 = (EditText) findViewById(R.id.ed_nbre2);
        txt_somme = (TextView) findViewById(R.id.txt_somme);


    }

    public void Calculer(View v) {

        // ajouter le teste de validite de donnees des edittexts
        // ajouter aussi la disponobilite de reseau

        new Webservice().execute(url_serv, ed_nbre1.getText().toString(), ed_nbre2.getText().toString());


    }

    class Webservice extends AsyncTask<String, Void, Integer> {
        @Override
        protected void onPreExecute() {
            super.onPreExecute();


            Toast.makeText(MainActivity.this, "Connexion au webservice ", Toast.LENGTH_SHORT).show();
        }

        @Override
        protected Integer doInBackground(String... params) {

            int somme = 0;

            try {
                URL url = new URL(params[0]);

                String par_nbre1 = params[1];
                String par_nbre2 = params[2];


                HttpURLConnection conx = (HttpURLConnection) url.openConnection();
                conx.setRequestMethod("GET");
                conx.setDoOutput(true);
                OutputStream os = conx.getOutputStream();
                BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(os));
                writer.write("var1=" + par_nbre1 + "&" + "var2=" + par_nbre2);
                writer.flush();
                writer.close();
                conx.connect();


                int reponse = conx.getResponseCode();

                if (reponse == HttpsURLConnection.HTTP_OK) {


                    InputStream is = conx.getInputStream();

                    BufferedReader br = new BufferedReader(new InputStreamReader(is));

                    String ligne = "", result = "";

                    while ((ligne = br.readLine()) != null) {

                        result += ligne;

                    }


                    JSONObject jo = new JSONObject(result);
                    somme = Integer.parseInt(jo.getString("somme"));
                    return somme;

                }


            } catch (MalformedURLException e) {
                e.printStackTrace();
            } catch (IOException e) {
                e.printStackTrace();
            } catch (JSONException e) {
                e.printStackTrace();
            }

            return null;


        }

        protected void onPostExecute(Integer i) {

            txt_somme.setText("" + i);


        }
    }
}

File: genererPDF/java/com/exemple/generPDF/MainActivity.kt
package com.example.genererpdf

import android.app.ProgressDialog
import android.os.AsyncTask
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*
import java.io.*
import java.net.HttpURLConnection
import java.net.MalformedURLException
import java.net.URL
import java.net.URLEncoder
import javax.net.ssl.HttpsURLConnection

class MainActivity : AppCompatActivity() {


    internal var adresse = "http://10.0.2.2/tutofpdf/genererPDF.php"
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        btn_genPDF.setOnClickListener(View.OnClickListener {


            if (!txt_nom.text.toString().equals("") && !txt_prenom.text.toString().equals("") && !txt_adr.text.toString().equals(""))
                webConnection().execute(txt_nom.text.toString(), txt_prenom.text.toString(), txt_adr.text.toString())
            else
                Toast.makeText(this,"Verifier vos donnnes ", Toast.LENGTH_LONG).show()
        })

    }

    internal inner class webConnection : AsyncTask<String, Void, String>() {
        lateinit var pb: ProgressDialog
        var reponse = 0

        override fun onPreExecute() {
            super.onPreExecute()

            pb = ProgressDialog(this@MainActivity)
            pb.setMessage("Connexion au site web ")

            pb.show()
            //  cancel(false) ;
        }


        override fun doInBackground(vararg params: String): String? {


            val nom = params[0]
            val prenom = params[1]
            val adress = params[2]


            var resultat = ""

            try {


                val url = URL(adresse)  // construire  l url a partir de url de type string


                val conx = url.openConnection() as HttpURLConnection



                conx.requestMethod = "POST"  //methode get par defaut
                conx.doInput = true  // par defaut
                conx.doOutput = true

                val OS = conx.outputStream
                val bufferedWriter = BufferedWriter(OutputStreamWriter(OS, "UTF-8"))
                val data = URLEncoder.encode("nom", "UTF-8") + "=" + URLEncoder.encode(nom, "UTF-8") + "&" +
                        URLEncoder.encode("prenom", "UTF-8") + "=" + URLEncoder.encode(prenom, "UTF-8") + "&" +
                        URLEncoder.encode("adresse", "UTF-8") + "=" + URLEncoder.encode(adress, "UTF-8")
                bufferedWriter.write(data)
                bufferedWriter.flush()
                bufferedWriter.close()
                OS.close()
                conx.connect()





                reponse = conx.responseCode
                if (reponse == HttpsURLConnection.HTTP_OK)
                // code 200 ok
                // code 404 not found
                {
                    val `is` = conx.inputStream
                    // methode avec des streams bufferises
                    val br = BufferedReader(InputStreamReader(`is`))

                    var ligne: String? = ""
                    ligne = br.readLine()
                    while (ligne != null) {

                        resultat += ligne
                        ligne = br.readLine()
                    }
                    return resultat
                }

            } catch (e: MalformedURLException) {
                e.printStackTrace()
            } catch (e: IOException) {
                e.printStackTrace()
            }

            return null
        }

        override fun onPostExecute(s: String) {
            super.onPostExecute(s)
            pb.dismiss()

            if (s == "OK") {
                Toast.makeText(this@MainActivity, "OK", Toast.LENGTH_LONG).show()

            } else
                Toast.makeText(this@MainActivity, "Erreur", Toast.LENGTH_LONG).show()


        }
    }
}
